modules/widgets/migrate-to-core/image-widget.php

```diff
@@ -202,6 +202,12 @@ function jetpack_migrate_image_widget() {
 
 		// We need to refresh on widgets page for changes to take effect.
 		add_action( 'current_screen', 'jetpack_refresh_on_widget_page' );
+	} else {
+		$widget_media_image = get_option( 'widget_media_image' );
+		if ( is_array( $widget_media_image ) ) {
+			delete_option( 'widget_image' );
+			Jetpack_Options::update_option( 'image_widget_migration', true );
```

Comment:
In this branch, if `widget_media_image` doesn't have data, `widget_image` won't be deleted.